app/controllers/PdfDownloadController.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.FrontendAppConfig
+import controllers.actions._
+import javax.inject.Inject
+import models.Case
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, Results}
+import service.{CasesService, FileService, PdfService}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.pdftemplates.applicationPdf
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class PdfDownloadController @Inject()(appConfig: FrontendAppConfig,
+                                      override val messagesApi: MessagesApi,
+                                      identify: IdentifierAction,
+                                      pdfService: PdfService,
+                                      caseService: CasesService,
+                                      fileService: FileService
+                                     ) extends FrontendController with I18nSupport {
+
+  def application(reference: String): Action[AnyContent] = identify.async { implicit request =>
+
+    caseService.getCaseForUser(request.eoriNumber, reference) flatMap {
+      case Some(c: Case) =>
+        fileService.getAttachmentMetadata(c) flatMap { attachmentData =>
+          pdfService.generatePdf(applicationPdf(appConfig, c, attachmentData)).map({ binaryFile =>
+            Results.Ok(binaryFile.content)
+              .as(binaryFile.contentType)
+              .withHeaders("Content-Disposition" -> s"attachment; filename=confirmation_$reference.pdf")
+          })
+        }
+      case _ => throw new Exception(s"Case ($reference) not found for user ${request.eoriNumber}")
```

Comment:
I think we can use the same approach of the operational UI (showing the "case not found error page") or alternatively this is ok, but we should not log the request to kibana because it contains PII (eori number)